### PR TITLE
[itbit, cexio, therock, bluetrade, bitfinex, gdax, quadriga] multiple improvements - mostly support for more parameters in getTradeHistory

### PR DESCRIPTION
**Pull request by npomfret**

### CODE REVIEW

**Comment by npomfret** [comment]
Not sure what's going on here.  The tests are passing for me.

**Comment by timmolter** [comment]
It looks like the dates in the test are off by one hour. The test is not local-timezone friendly I guess.

**Comment by npomfret** [comment]
Ah! Ok, thanks.  (Maybe thats why the original implementation left them out!)

**Comment by timmolter** [comment]
there's a way to properly do it, but I forgot. I'm 99% sure some other tests do this properly.

**Comment by timmolter** [comment]
I think you have to set the timezone on your Formatter. See other examples. (I'm recalling from memory, could be wrong)

**Comment by npomfret** [comment]
How often is the snap-shot built?

**Comment by timmolter** [comment]
On every commit or accepted pull request. 

I set up Travis CI to do it.

**Comment by npomfret** [comment]
Thanks.

File: xchange-quadrigacx/src/main/java/org/knowm/xchange/quadrigacx/service/QuadrigaCxTradeServiceRaw.java [hunk 1]
```diff
@@ -64,10 +63,16 @@ public boolean cancelQuadrigaCxOrder(String orderId) throws IOException {
         orderId);
   }
 
-  public QuadrigaCxUserTransaction[] getQuadrigaCxUserTransactions(CurrencyPair currencyPair, Long numberOfTransactions) throws IOException {
-
-    return quadrigacxAuthenticated.getUserTransactions(QuadrigaCxUtils.currencyPairToBook(currencyPair),
-        exchange.getExchangeSpecification().getApiKey(), signatureCreator, exchange.getNonceFactory(), numberOfTransactions);
+  public QuadrigaCxUserTransaction[] getQuadrigaCxUserTransactions(CurrencyPair currencyPair, Long numberOfTransactions, long offset, String sort) throws IOException {
```
Comment (andre77) [inline xchange-quadrigacx/src/main/java/org/knowm/xchange/quadrigacx/service/QuadrigaCxTradeServiceRaw.java]: What is the purpose of having two identical methods here ???
the only difference, i see, is the "offset" perameter being long/Long

Comment (npomfret) [inline xchange-quadrigacx/src/main/java/org/knowm/xchange/quadrigacx/service/QuadrigaCxTradeServiceRaw.java]: that's not intentional, thanks for pointing it out

Comment (andre77) [inline xchange-quadrigacx/src/main/java/org/knowm/xchange/quadrigacx/service/QuadrigaCxTradeServiceRaw.java]: create a PR for that already
https://github.com/timmolter/XChange/pull/1604

**Pull request merged**